src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -0,0 +1,85 @@
+namespace CarbonAware.WepApi.IntegrationTests;
+
+using CarbonAware.DataSources.Configuration;
+using CarbonAware.WebApi.IntegrationTests;
+using NUnit.Framework;
+using System.Net;
+using System.Web;
+
+
+/// <summary>
+/// Tests that the Web API controller handles and packages various responses from a plugin properly 
+/// including empty responses and exceptions.
+/// </summary>
+[TestFixture(DataSourceType.JSON)]
+[TestFixture(DataSourceType.WattTime)]
+public class CarbonAwareControllerTests : IntegrationTestingBase
+{
+    private string healthURI = "/health";
+    private string fakeURI = "/fake-endpoint";
+    private string bestLocationsURI = "/emissions/bylocations/best";
+
+    public CarbonAwareControllerTests(DataSourceType dataSource) : base(dataSource) { }
+
+    [Test]
+    public async Task HealthCheck_ReturnsOK()
+    {
+        //Use client to get endpoint
+        var result = await _client.GetAsync(healthURI);
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+    }
+
+    [Test]
+    public async Task FakeEndPoint_ReturnsNotFound()
+    {
+        var result = await _client.GetAsync(fakeURI);
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.NotFound));
+    }
+
+    //ISO8601: YYYY-MM-DD
+    [TestCase("2022-1-1", 1, "eastus")]
+    [TestCase("2021-12-25", 1, "westus")]
+    public async Task BestLocations_ReturnsOK(DateTime start, int offset, string location)
+    {
+        var end = start.AddDays(offset);
+
+        //Sets up any data endpoints needed for mocking purposes
+        _dataSourceMocker.SetupDataMock(start, end, location);
+
+        //Call the private method to construct with parameters
+        var endpointURI = ConstructBestLocationsURI(location, start, end);
+
+        //Get response and response content
+        var result = await _client.GetAsync(endpointURI);
+        var resultContent = await result.Content.ReadAsStringAsync();
+
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+        Assert.That(resultContent, Is.Not.Null);
+    }
+
+    private string ConstructBestLocationsURI(string location, DateTime start, DateTime end)
```

Comment:
Consider cribbing from [this method](https://github.com/microsoft/carbon-aware-sdk/blob/870ac27bdb26d2170479b6a98175f2b4595533c6/src/CarbonAware.Tools/CarbonAware.Tools.WattTimeClient/src/WattTimeClient.cs#L308-L331) and making this available as a public function in the base class